module/zfs/ddt.c

```diff
@@ -816,14 +816,14 @@ ddt_entry_compare(const void *x1, const void *x2)
 {
 	const ddt_entry_t *dde1 = x1;
 	const ddt_entry_t *dde2 = x2;
-	const uint64_t *u1 = (const uint64_t *)&dde1->dde_key;
-	const uint64_t *u2 = (const uint64_t *)&dde2->dde_key;
+	const ddt_key_cmp_t *u1 = (ddt_key_cmp_t *)&dde1->dde_key;
+	const ddt_key_cmp_t *u2 = (ddt_key_cmp_t *)&dde2->dde_key;
 	int i;
 
 	for (i = 0; i < DDT_KEY_WORDS; i++) {
-		if (u1[i] < u2[i])
+		if (u1->dkc_key[i] < u2->dkc_key[i])
 			return (-1);
-		if (u1[i] > u2[i])
+		if (u1->dkc_key[i] > u2->dkc_key[i])
 			return (1);
 	}
```

Comment:
What @richardelling is saying is that comparison will most likely be resolved by looking at first word of the key. However, the method can be made significantly faster by computing the result directly, instead of using conditionals. See PR #5033